### PR TITLE
Attributed string

### PR DESCRIPTION
**Pull request by mstokercricut**
* Adds translation of AttributedString, AttributedSubstring, AttributeContainer, and a notion of AttributeScope to Kotlin
* Adds translation of URL to Kotlin
* Provides testing for these in TestAPI

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ func wasmIncompatible<T>(_ things: @autoclosure () -> [T]) -> [T] {
 
 let package = Package(
     name: "FishyJoes",
-    platforms: [.macOS(.v11)],
+    platforms: [.macOS(.v12), .iOS(.v15)],
```
Comment (cobbal) [inline Package.swift:29]: Do we support iOS in any way? (Maybe we will need to some day for flutter)

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    let containedNamedTypes: [TranslatedType] = []
+    let neutralName = "AttributedSubstring.Runs.Run"
+    let nodeName = "AttributedSubstring.Runs.Run"
+    let kotlinPackage: String? = "com.cricut.fishyjoes.runtime"
+    let kotlinName = "AttributedSubstring.Runs.Run"
+    let jniType = JNIType.object("com/cricut/fishyjoes/runtime/AttributedString$Runs$Run")
+    let cSharpType = CSharpClass.CSType.named(package: "Cricut.FishyJoesRuntime", name: "AttributedString.Runs.Run")
+    let cppName = "AttributedString::Runs::Run"
+    let definingModule = Module.runtime
+}
```
Comment (cobbal) [inline Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift:157]: Do these need to be all different types? They seem very similar, and could possibly be just different instances of one type to reduce duplication.

Comment (cobbal) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift:157]: Similar to, but probably not the same as PrimitiveType

File: Sources/FishyJoesJavaRuntime/AttributedString/AttributeScope+java.swift [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            self.link = container.link
+            self.languageIdentifier = container.languageIdentifier
+        }
+    }
+}
+
+extension AttributeContainer {
+    public init(_ foundationAttributes: FoundationAttributes) {
+        self = AttributeContainer()
+        if let link = foundationAttributes.link { self.link = link }
```
Comment (cobbal) [inline Sources/FishyJoesJavaRuntime/AttributedString/AttributeScope+java.swift]: nit: I kept trying to read this as trailing closure syntax. I think I don't like one-liner if-lets

**Review by cobbal — APPROVED** [review]
Approved with nits

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift (hunk 2 above)
Comment (mstokercricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift:157]: I'll take a look, but I think they do. Each type has pretty different values for each field. I can drop the AttributeScope type, though. The merge policy type could go, too.

File: Package.swift (hunk 1 above)
Comment (mstokercricut) [reply Package.swift:29]: That is one place. I also want to know if iOS lacks support for anything here, even if we don't directly support it, as it informs us where inconsistencies in interface might pop up.

File: Sources/FishyJoesJavaRuntime/AttributedString/AttributeScope+java.swift (hunk 3 above)
Comment (mstokercricut) [reply Sources/FishyJoesJavaRuntime/AttributedString/AttributeScope+java.swift]: 👍🏿

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift (hunk 2 above)
Comment (mstokercricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedAttributedString.swift:157]: Removed in favor of a boolean.

**Pull request merged**